tests/test_core.py

```diff
@@ -1118,6 +1120,67 @@ class MyClass(NSObject):
             class MyClass(NSObject):  # noqa: F811
                 pass
 
+    def test_class_auto_rename(self):
+        """Test automatic renaming system of ObjCClass."""
+
+        # Check that the global option works.
+
+        ObjCClass.auto_rename = True
+
+        class TestClass(NSObject):
+            @objc_method
+            def oldMethod(self):
+                pass
+
+        testclass1 = TestClass
+
+        class TestClass_2(NSObject):
+            pass
+
+        class TestClass(NSObject):  # noqa: F811
+            @objc_method
+            def testMethod(self):
+                return "TEST1"
+
+        self.assertEqual(TestClass.name, "TestClass_3")
+        self.assertIsNot(testclass1, TestClass)
+
+        object1 = TestClass.new()
+
+        with self.assertRaises(AttributeError):
+            object1.oldMethod()
+
+        self.assertEqual(object1.testMethod(), "TEST1")
+
+        # Check that the per-class option works.
```

Comment:
This test is complex enough that it would be worth splitting into two parts - one for the global setting; and one for the class-specific setting. As part of that, we probably want to make the test class name something that is unique enough that we won't accidentally re-use the same name elsewhere, causing confusion (e.g. call the class TestRenamedClass, rather than TestClass)